src/main/java/hudson/plugins/s3/callable/S3UploadCallable.java

```diff
@@ -33,9 +34,10 @@
     private final boolean produced;
     private final boolean useServerSideEncryption;
 
-    public S3UploadCallable(boolean produced, String accessKey, Secret secretKey, boolean useRole, Destination dest, List<MetadataPair> userMetadata, String storageClass,
+    public S3UploadCallable(boolean produced, String accessKey, Secret secretKey, boolean useRole, String bucketName, Destination dest, List<MetadataPair> userMetadata, String storageClass,
```

Comment:
I don't know any plugin that have a dependency on s3-plugin. But it might exist one.